src/show.jl

```diff
@@ -144,38 +150,42 @@ function printcoefficient(io::IO, pj::Complex{T}, j, mimetype) where {T}
     elseif hasimag
         b = imag(pj)
         (showone(T) || b != one(T)) && show(io,  mimetype, b)
-        (isnan(imag(pj)) || isinf(imag(pj))) && print(io, showop(mimetype, "*"))
+        (isnan(b) || isinf(b)) && print(io, showop(mimetype, "*"))
         show(io, mimetype, im)
     else
         return
     end
 end
-
-
-## show a single term
 function printcoefficient(io::IO, pj::T, j, mimetype) where {T}
-    pj == one(T) && !(showone(T) || j == 0) && return
-    show(io, mimetype, pj)
+    # A simple heuristic to determine when to parenthesize
+    # coefficients of various types, e.g. dual numbers
+    if contains(string(pj), " + ") || contains(string(pj), " - ")
```

Comment:
Should this be "+" and "-" in case the type is not generous with space?

Reply:
Yes, the only real change is the heuristic,

Perhaps, that is a good idea, that would insert brackets  around the `c` in the following situation

```julia
using SymPy
@vars a b c
Polynomials.Poly([a+b, -c, b+c])
``'

`Poly((a + b) + -c*x + (b + c)*x^2)`

Don't know if it is nicer with or without? I guess one would like a more elaborate scheme to handle this like other real numbers are handled. The pull request only fixes the bare minimum so that the expressions are correct.